packages/hadron-document/src/utils.ts

```diff
@@ -70,6 +76,12 @@ function makeEJSONIdiomatic(value: EJSON.SerializableTypes): void {
       }
       continue;
     }
+    if (entry.$date && entry.$date.$numberLong) {
+      const number = entry.$date.$numberLong;
+      if (number >= 0 && number <= maxFourYearDate) {
+        entry.$date = new Date(+number);
```

Comment:
Should this be

```suggestion
        entry.$date = new Date(+number).toISOString();
```

?

Reply:
(also, you may want to add tests for this)

Reply:
I don't know. It seems to work. I suppose I can just add that to be safe.

Reply:
Yeah I see this file has no tests yet :(

Reply:
It does: https://github.com/mongodb-js/compass/blob/03211164e94cb45075ead41b2090cacd99a71301/packages/hadron-document/test/document.test.ts#L2317-L2382

Reply:
Added now.